ui/pages/settings/networks-tab/custom-content-search/custom-content-search.test.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { shallow, mount } from 'enzyme';
+import TextField from '../../../../components/ui/text-field';
+import CustomContentSearch from './custom-content-search';
+
+describe('CustomContentSearch', () => {
+  let wrapper;
+
+  it('should render custom content search correctly', () => {
+    wrapper = shallow(
+      <CustomContentSearch onSearch={() => undefined} networksList={[]} />,
+      {
+        context: {
+          t: (s) => `${s}`,
+        },
+      },
+    );
+
+    expect(wrapper.find(TextField).props().id).toStrictEqual('search-networks');
+    expect(wrapper.find(TextField).props().value).toBeUndefined();
+  });
+
+  it('should call onSearch prop with input value', () => {
+    const onSearch = jest.fn();
+    wrapper = mount(
+      <CustomContentSearch
+        onSearch={onSearch}
+        networksList={[]}
+        searchQueryInput="Avalanche"
+      />,
+    );
+    wrapper.find('input').simulate('change');
```

Comment:
I'm wondering if you really need to fire this event manually. It's my understanding that `onChange` will fire automatically on an input element when its `value` changes. React Testing Library provides a `rerender` function. What happens if you render the component once without any `searchQueryInput` and then again with an updated `searchQueryInput`? You can read about `rerender` here: https://testing-library.com/docs/react-testing-library/api#rerender

Reply:
@mcmire Done [here](https://github.com/MetaMask/metamask-extension/pull/13560/commits/c48e1890932565addc62ccd37f3dfad1a324c708)